### PR TITLE
Dp 22990/update documentation for new workflow

### PR DESCRIPTION
**Pull request by clairesunstudio**
Hi Mass.gov Developers,

We are making some changes to our Mayflower-to-Openmass integration workflow. If you are working on a feature development that requires both Mayflower and Openmass changes, or you are part of the release process, these changes will impact your work. 

Openmass is going to continuously integrate Mayflower by always pointing to the Mayflower Artifacts built from the Mayflower develop branch. (See [the Openmass Mayflower doc](https://github.com/massgov/openmass/blob/develop/docs/mayflower.md) for more information) This means Openmass and Mayflower releases are not going to be in sync anymore — Openmass can be released with the latest Mayflower develop branch without a Mayflower release; Mayflower changelogs will not be compiled at the time of the Openmass release, but they are still required as they will be compiled during Mayflower releases.
> If you are curious about how we landed with this decision, please refer to the these JIRA tickets: [DP-22990](https://massgov.atlassian.net/browse/DP-22990) and [DP-22926](https://massgov.atlassian.net/browse/DP-22926).

## For developers
All openmass feature development PRs that require Mayflower changes must pass all these checks before squashing and merging into develop:
- [ ] PR reviewed and approved
- [ ] The corresponding Mayflower PR is merged into develop
- [ ]  Mayflower artifacts is pointing to the dev-develop version in composer.json
- [ ]  Up-to-date with latest openmass develop branch (resolve any composer.lock conflicts with the develop branch)
- [ ]  All circleCI tests are green on the branch

**NEW:** For a Mayflower-only change, you will need to create an openmass PR that links to your Mayflower change.
Because we are no longer compiling Mayflower changelogs for each Openmass release, we expect you to add a changelog in each repo; follow the branch naming conventions and include the ticket number in the branch names so they appear in JIRA. Please also link to the corresponding Mayflower PR in the Openmass PR description. Doing so will help others review the PRs together and find the related PRs quickly in case where a hotfix is needed.

> Refer to [Mayflower integration peer review checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md#mayflower-integration)

## For release managers
A nightly deployment to the CD environment will automatically kick off at 11PM EST. CircleCI will build the Openmass develop branch with the Mayflower Artifacts generated from the latest Mayflower develop branch. Visual regression tests (Backstop) and 500 error tests (Nightcrawler) will also be run on the CD environment every night. 

You will no longer need to create a mayflower integration PR on the release day.  Please refer to the [updated CI/CD and release day documentation](https://github.com/massgov/massgov-internal-docs/blob/master/releases/mayflower-ci-cd.md) for the review process. 

> To view a detailed diagram of this new workflow, go to https://miro.com/app/board/o9J_lsvGP9M=/?invite_link_id=257854526758

### CODE REVIEW

File: docs/mayflower.md [hunk 1]
```diff
@@ -1,50 +1,67 @@
...
+
+If you have committed your change to point to a feature branch of mayflower artifacts instead of the develop branch, make sure you change it back to point to `dev-develop` before merging into develop. 
+
+#### Merging a feature PR with Mayflower changes:
+All openmass feature development PRs that require Mayflower changes must pass all these checks before squashing and merging into develop:
+- [ ] PR reviewed and approved 
+- [ ] Corresponding Mayflower PR is merged into develop
+- [ ] Mayflower artifacts is pointing to the `dev-develop` version in composer.json
+- [ ] Up-to-date with latest openmass develop branch (resolve any composer.lock conflicts with the develop branch)
+- [ ] All circleCI tests are green on your branch
```
Comment (weitzman) [inline docs/mayflower.md]: This content might be more appropriate for the peer review checklist, which is linked at bottom of every PR. But we can do that in a followup.

File: docs/mayflower.md [hunk 2]
```diff
@@ -1,50 +1,67 @@
...
+
+Because we are no longer compiling Mayflower changelogs for each Openmass release, we expect you to follow the PR naming conventions, include the ticket number in the branch names so they appear in JIRA. Please also link to the corresponding Mayflower PR in the Openmass PR description. Doing so will help others review the PRs together and find the related PRs quickly in case where a hotfix is needed.
```
Comment (weitzman) [inline docs/mayflower.md]: In Openmass, we have a branch naming convention, but not PR naming convention AFAIK.

Comment (clairesunstudio) [reply docs/mayflower.md]: 496a3f0

File: docs/mayflower.md (hunk 1 above)
Comment (clairesunstudio) [reply docs/mayflower.md]: make sense, I moved the checklist into the peer review doc 4160130

**Pull request merged**